### PR TITLE
Try with resources in example

### PR DESCRIPTION
**Pull request by DotSpy**
Create basic span example should be with try with resources

### CODE REVIEW

File: QUICKSTART.md [hunk 1]
```diff
@@ -47,9 +47,11 @@ Tracer tracer =
 To create a basic span, you only need to specify the name of the span.
 The start and end time of the span is automatically set by the OpenTelemetry SDK.
 ```java
-Span span = tracer.spanBuilder("SpanName").startSpan();
+Span span = tracer.spanBuilder("my span").startSpan();
+try (Scope scope = tracer.withSpan(span)) {
 // your use case
 ...
```
Comment (arminru) [inline QUICKSTART.md]:
```suggestion
	// your use case
	...
```

File: QUICKSTART.md [hunk 2]
```diff
@@ -47,9 +47,11 @@ Tracer tracer =
...
+}
 span.end();
```
Comment (arminru) [inline QUICKSTART.md]:
```suggestion
span.end(); // closing the scope does not end the span, this has to be done manually
```

File: QUICKSTART.md [hunk 3]
```diff
@@ -47,10 +47,12 @@ Tracer tracer =
...
-Span span = tracer.spanBuilder("SpanName").startSpan();
-// your use case
-...
-span.end();
+Span span = tracer.spanBuilder("my span").startSpan();
+try (Scope scope = tracer.withSpan(span)) {
+	// your use case
+	...
+}
+span.end(); // closing the scope does not end the span, this has to be done manually
```
Comment (bogdandrutu) [inline QUICKSTART.md]: I would put this in a `finally` block to ensure always called. Also we may have a `catch` block where we set a `status`

**Comment by DotSpy** [comment]
Should we add comments to catch block or all ok with it?

File: QUICKSTART.md (hunk 3 above)
Comment (DotSpy) [reply QUICKSTART.md]: I wanted to do it at start, but name of section (### Create basic Span) confused me

File: QUICKSTART.md [hunk 4]
```diff
@@ -47,10 +47,16 @@ Tracer tracer =
...
-Span span = tracer.spanBuilder("SpanName").startSpan();
-// your use case
-...
-span.end();
+Span span = tracer.spanBuilder("my span").startSpan();
+try (Scope scope = tracer.withSpan(span)) {
+	// your use case
+	...
+} catch (Throwable t) {
+    Status status = Status.UNKNOWN.withDescription("Change it to your error message");
```
Comment (arminru) [inline QUICKSTART.md]:
```suggestion
    Status status = Status.UNKNOWN.withDescription(t.getMessage());
```

File: QUICKSTART.md [hunk 5]
```diff
@@ -51,8 +51,12 @@ Span span = tracer.spanBuilder("my span").startSpan();
 try (Scope scope = tracer.withSpan(span)) {
 	// your use case
 	...
```
Comment (thisthat) [inline QUICKSTART.md]:
```suggestion
	...
	span.setStatus(Status.OK);
```
Maybe add also this so the reader does not get confused and think status is set only on error.

Comment (DotSpy) [reply QUICKSTART.md]: @thisthat isn't it default when we end?

Comment (arminru) [reply QUICKSTART.md]: You only have to call setStatus if you want to override the default, which already is OK.

File: QUICKSTART.md (hunk 4 above)
Comment (DotSpy) [reply QUICKSTART.md]: @arminru i'm not sure about it, t.getMessage() can provide a little explanation and user may want to change this to custom string including or excluding t.getMessage or even not exposing inner error

Comment (arminru) [reply QUICKSTART.md]: Alright, I'm fine either way.

File: QUICKSTART.md (hunk 5 above)
Comment (thisthat) [reply QUICKSTART.md]: Oh, you are right!

**Pull request merged**